compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -136,8 +136,9 @@ object desugar {
     )
 
   /** A derived type definition watching `sym` */
-  def derivedTypeParam(sym: TypeSymbol)(using Context): TypeDef =
-    TypeDef(sym.name, DerivedFromParamTree().watching(sym)).withFlags(TypeParam)
+  def derivedTypeParamWithVariance(sym: TypeSymbol)(using Context): TypeDef =
+    val variance = VarianceFlags & sym.flags
+    TypeDef(sym.name, DerivedFromParamTree().watching(sym)).withFlags(TypeParam | Synthetic | variance)
```

Comment:
For consistency, we should probably set Synthetic on the other derivedTypeParam and on derivedTermParam assuming that doesn't break things.

Reply:
If I add synthetic flag in derivedTermParam it changes the positions in this test: https://github.com/lampepfl/dotty/blob/ce48f5a2c79373f8a33ced1d251de0206ed5cba9/language-server/test/dotty/tools/languageserver/DefinitionTest.scala#L182

Reply:
yeah so maybe it's not worth it, what does it change the position to?

Reply:
from line 0-15:16 to line 0-11:14

Reply:
is this worth investigating?

Reply:
Probably not